modules/light-clients/08-wasm/types/client_state_test.go

```diff
@@ -938,217 +938,137 @@ func (suite *TypesTestSuite) TestVerifyNonMembershipGrandpa() {
+	var (
+		clientState      exported.ClientState
+		expClientStateBz []byte
+		path             exported.Path
+		proof            []byte
+		proofHeight      exported.Height
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expError error
+	}{
+		{
+			"success",
+			func() {
+				expClientStateBz = suite.store.Get(host.ClientStateKey())
+				suite.mockVM.RegisterSudoCallback(types.VerifyNonMembershipMsg{}, func(_ wasmvm.Checksum, _ wasmvmtypes.Env, sudoMsg []byte, _ wasmvm.KVStore,
+					_ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction,
+				) (*wasmvmtypes.Response, uint64, error) {
+					var payload types.SudoMsg
+					err := json.Unmarshal(sudoMsg, &payload)
+					suite.Require().NoError(err)
+
+					suite.Require().NotNil(payload.VerifyNonMembership)
+					suite.Require().Nil(payload.CheckSubstituteAndUpdateState)
+					suite.Require().Nil(payload.UpdateState)
+					suite.Require().Nil(payload.UpdateStateOnMisbehaviour)
+					suite.Require().Nil(payload.VerifyMembership)
+					suite.Require().Nil(payload.VerifyUpgradeAndUpdateState)
+					suite.Require().Equal(proofHeight, payload.VerifyNonMembership.Height)
+					suite.Require().Equal(path, payload.VerifyNonMembership.Path)
+					suite.Require().Equal(proof, payload.VerifyNonMembership.Proof)
+
+					bz, err := json.Marshal(types.EmptyResult{})
+					suite.Require().NoError(err)
+
+					return &wasmvmtypes.Response{Data: bz}, types.DefaultGasUsed, nil
+				})
+			},
+			nil,
+		},
+		{
+			"success: with update client state",
+			func() {
+				suite.mockVM.RegisterSudoCallback(types.VerifyNonMembershipMsg{}, func(_ wasmvm.Checksum, _ wasmvmtypes.Env, sudoMsg []byte, store wasmvm.KVStore,
+					_ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction,
+				) (*wasmvmtypes.Response, uint64, error) {
+					var payload types.SudoMsg
+					err := json.Unmarshal(sudoMsg, &payload)
+					suite.Require().NoError(err)
+
+					suite.Require().NotNil(payload.VerifyNonMembership)
+					suite.Require().Nil(payload.CheckSubstituteAndUpdateState)
+					suite.Require().Nil(payload.UpdateState)
+					suite.Require().Nil(payload.UpdateStateOnMisbehaviour)
+					suite.Require().Nil(payload.VerifyMembership)
+					suite.Require().Nil(payload.VerifyUpgradeAndUpdateState)
+					suite.Require().Equal(proofHeight, payload.VerifyNonMembership.Height)
+					suite.Require().Equal(path, payload.VerifyNonMembership.Path)
+					suite.Require().Equal(proof, payload.VerifyNonMembership.Proof)
+
+					bz, err := json.Marshal(types.EmptyResult{})
+					suite.Require().NoError(err)
+
+					expClientStateBz = []byte("client-state-data")
+					store.Set(host.ClientStateKey(), expClientStateBz)
+
+					return &wasmvmtypes.Response{Data: bz}, types.DefaultGasUsed, nil
+				})
+			},
+			nil,
+		},
+		{
+			"wasm vm returns invalid proof error",
+			func() {
+				proof = []byte("invalid proof")
```

Comment:
Is the value of proof ever used anywhere? If not what is the point of this instead of using a constant?

Reply:
True, this assignment doesn't really make much of a difference in this case, but the var is exposed at the top of the test func so that we can do the assertion in success cases that the go layer has passed it around without mutating it.